patches/src/main/kotlin/app/revanced/patches/spotify/misc/lyrics/ChangeLyricsProviderPatch.kt

```diff
@@ -0,0 +1,148 @@
+package app.revanced.patches.spotify.misc.lyrics
+
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.fingerprint
+import app.revanced.patcher.patch.bytecodePatch
+import app.revanced.patcher.patch.stringOption
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.patches.spotify.misc.extension.IS_SPOTIFY_LEGACY_APP_TARGET
+import app.revanced.util.getReference
+import app.revanced.util.indexOfFirstInstruction
+import app.revanced.util.indexOfFirstInstructionOrThrow
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.instruction.BuilderInstruction35c
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+import com.android.tools.smali.dexlib2.immutable.reference.ImmutableMethodReference
+import java.net.InetAddress
+import java.net.URI
+import java.net.URISyntaxException
+import java.net.UnknownHostException
+import java.util.logging.Logger
+
+
+@Suppress("unused")
+val changeLyricsProviderPatch = bytecodePatch(
+    name = "Change lyrics provider",
+    description = "Changes the lyrics provider to a custom one.",
+) {
+    compatibleWith("com.spotify.music")
+
+    val lyricsProviderUrl by stringOption(
+        key = "lyricsProviderUrl",
+        title = "Lyrics provider URL",
+        description = "The URL to a custom lyrics provider.",
+        required = true,
+        default = "lyrics.natanchiodi.fr"
```

Comment:
Since the option asks for an URL
```suggestion
        default = "https://lyrics.natanchiodi.fr"
```

Reply:
The patch isn't specific to this repository

Reply:
> Any protocol entered here is ignored.

Yes, however the option asks for an URL and a URL asks for a protocol. 

> What if the user enters something stupid like:
> ftp://lyrics.natanchiodi.fr
> The patch will happily accept and use it as-if nothing is wrong.

Nothing is wrong. The option asked for an URL, the user entered one. The host is extracted from it.

> Or what if they thing they can use a local server without https

The lyrics backend setups https. The user can enter HTTP or https regardless of that for the reason.
The option asks for an URL so users might assume the protocol will be respected. You can add to the description, that the host only matters.